float32 x
float32 y
---
float32 next_x
float32 next_y 
